apple/Sources/FerrostarSwiftUI/Views/GridViews/NavigatingInnerGridView.swift

```diff
@@ -1,6 +1,23 @@
 import FerrostarCore
 import SwiftUI
 
+/// Determines which camera control is shown in the navigation inner grid view.
+public enum CameraControlState {
+    /// Don't show any camera control.
+    case hidden
+
+    /// Shows the recenter button.
+    ///
+    /// The action is responsible for resetting the camera
+    /// to a state that follows the user.
+    case showRecenter(() -> Void)
+
+    /// Shows the route overview button.
+    ///
+    /// The action is responsible for transitioning the camera to an overview of the route.
+    case showRouteOverview(() -> Void)
+}
```

Comment:
This is the first significant design decision. I think that an enum makes the camera control states significantly cleaner to express. We can add more later if necessary, but the common ones are here.